scenarios/sslstream.benchmarks.yml

```
@@ -0,0 +1,99 @@
+# Example:
+# crank --config https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/sslstream.benchmarks.yml --scenario read-write --profile aspnet-perf-lin
+
+imports:
+  - https://raw.githubusercontent.com/aspnet/Benchmarks/main/scenarios/aspnet.profiles.standard.yml
+
+variables:
+  # common parameters
+  serverPort: 5000
+  sendBufferSize: 32768 # bytes
+  receiveBufferSize: 32768 # bytes
+  sslProtocols: "None" # <Default|None|Ssl2|Ssl3|Tls|Tls11|Tls12|Tls13>
+  certSource: "Context" # <Callback|Certificate|Context>

```

Is "certificate source" a term that is commonly used? 😅 because this option was confusing me, for quite some time I thought it was about where the _benchmark app_ should get the cert 😅

---

We use something similar in S.N.S unit tests

https://github.com/dotnet/runtime/blob/77d643d8678d0cae1144557f106bb693a1cbca25/src/libraries/System.Net.Security/tests/FunctionalTests/SslStreamMutualAuthenticationTest.cs#L36-L41

I am open for suggestions.

---

I was thinking about it, and maybe something like "CertSelection"/"CertSelectionType"?

and this is only for the server now? you don't pass it to the client, but I see there's an option to do that in the future. then it might make sense to separate the variables by name, because the selection can be different between client and server.

```suggestion
  ServerCertSelection: "CertContext" # <Callback|Property|CertContext>
  ClientCertSelection: "CertContext" # <Callback|Collection|CertContext>
```